### PR TITLE
Merge CLIs into one with subcommands

### PR DESCRIPTION
**Pull request by rlustin**

### CODE REVIEW

File: src/cli.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: dj-library-gain-calculator
+version: "0.1.0"
+about: Analyses all tracks in a Traktor DJ library to have constant loudness.
+subcommands:
+    - library:
```
Comment (padenot) [inline src/cli.yml]: Call this collection, it's the term traktor uses

File: src/cli.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+              allow_hyphen_values: true
+              default_value: "-14.0"
+          - write:
+              help: Updates the Traktor library in place.
+              short: w
+              long: write
+              global: true
+              conflicts_with:
+                - output
+    - track:
```
Comment (padenot) [inline src/cli.yml]: call this `scanner`, it's what it's for

File: src/library.rs [hunk 3]
```diff
@@ -1,28 +1,14 @@
...
-    match process(app) {
-        Ok(_) => {}
-        Err(error) => exit_with_error(&error.to_string()),
-    };
-}
-
-fn process(app: App) -> Result<(), AppError> {
-    let matches = app.get_matches();
-
+pub fn run(matches: &ArgMatches) -> Result<(), AppError> {
```
Comment (padenot) [inline src/library.rs]: rename this file accordingly

File: src/track.rs [hunk 4]
```diff
@@ -1,23 +1,11 @@
...
-        Ok(_) => {}
-        Err(error) => exit_with_error(&error.to_string()),
-    };
-}
-
-fn process(app: App) -> Result<(), AppError> {
-    let matches = app.get_matches();
+use crate::analysis::scan_loudness;
+use crate::error::AppError;
+use crate::utils::linear_to_db;
```
Comment (padenot) [inline src/track.rs]: rename this file accordingly

File: tests/integration_test.rs [hunk 5]
```diff
@@ -7,7 +7,10 @@ use tempdir::TempDir;
 fn it_fails_when_file_does_not_exist() -> Result<(), Box<dyn std::error::Error>> {
     let mut command = Command::cargo_bin("dj-library-gain-calculator")?;
 
-    command.arg("--input").arg("tests/vectors/not_found.nml");
+    command
+        .arg("library")
```
Comment (padenot) [inline tests/integration_test.rs]: and this

File: tests/integration_test.rs [hunk 6]
```diff
@@ -27,6 +30,7 @@ fn it_processes_a_1_element_collection_to_stdout() -> Result<(), Box<dyn std::er
     input_content.truncate(input_content.len() - 1);
 
     command
+        .arg("library")
```
Comment (padenot) [inline tests/integration_test.rs]: and this

File: tests/integration_test.rs [hunk 7]
```diff
@@ -45,6 +49,7 @@ fn it_processes_a_1_element_collection_to_a_file() -> Result<(), Box<dyn std::er
     let output_path = output_dir.path().join("output.nml");
 
     command
+        .arg("library")
```
Comment (padenot) [inline tests/integration_test.rs]: and this

**Review by padenot — DISMISSED** [review]

File: src/cli.yml (hunk 1 above)
Comment (rlustin) [reply src/cli.yml]: Fixed.

File: src/cli.yml (hunk 2 above)
Comment (rlustin) [reply src/cli.yml]: Fixed.

File: src/library.rs (hunk 3 above)
Comment (rlustin) [reply src/library.rs]: Fixed.

File: src/track.rs (hunk 4 above)
Comment (rlustin) [reply src/track.rs]: Fixed.

File: tests/integration_test.rs (hunk 5 above)
Comment (rlustin) [reply tests/integration_test.rs]: Fixed.

File: tests/integration_test.rs (hunk 6 above)
Comment (rlustin) [reply tests/integration_test.rs]: Fixed.

File: tests/integration_test.rs (hunk 7 above)
Comment (rlustin) [reply tests/integration_test.rs]: Fixed.

**Review by padenot — APPROVED** [review]

**Pull request merged**